### PR TITLE
jerryx: root handle scope convenience methods

### PR DESCRIPTION
**Pull request by legendecas**
Series convenience methods of root handle scope.

Closes root handle scope on iotjs exit.

- [x] tests
- [x] add jerry-ext dependency to iotjs (#247 )
- [x] build option to include jerryx (#247 )

### CODE REVIEW

File: cmake/jerry.cmake [hunk 1]
```diff
@@ -32,7 +32,7 @@ ExternalProject_Add(hostjerry
     -DJERRY_LIBC=OFF
     -DJERRY_CMDLINE=OFF
     -DJERRY_CMDLINE_SNAPSHOT=ON
-    -DJERRY_EXT=OFF
+    -DJERRY_EXT=ON
```
Comment (yorkie) [inline cmake/jerry.cmake]: Let's open it once the full functions are complete :)

File: cmake/jerry.cmake [hunk 2]
```diff
@@ -52,7 +52,7 @@ macro(add_cmake_arg TARGET_ARG KEY)
 endmacro(add_cmake_arg)
 
 # Target libjerry
-set(JERRY_LIBS jerry-core jerry-port-default)
+set(JERRY_LIBS jerry-core jerry-port-default jerry-ext)
```
Comment (yorkie) [inline cmake/jerry.cmake]: We should set this by checking the define `JERRY_EXT`.

File: cmake/jerry.cmake [hunk 3]
```diff
@@ -138,6 +138,7 @@ ExternalProject_Add(libjerry
 set_property(DIRECTORY APPEND PROPERTY
   ADDITIONAL_MAKE_CLEAN_FILES
     ${CMAKE_BINARY_DIR}/lib/libjerry-core.a
+    ${CMAKE_BINARY_DIR}/lib/libjerry-ext.a
```
Comment (yorkie) [inline cmake/jerry.cmake]: Ditto

File: src/iotjs.c [hunk 4]
```diff
@@ -26,6 +26,7 @@
 #endif
 #include "jerryscript-port.h"
 #include "jerryscript.h"
+#include "jerryscript-ext/handle-scope.h"
```
Comment (yorkie) [inline src/iotjs.c]: Also, we are not starting using HandleScope yet, so adding this by MACRO would be reasonable :)

File: src/iotjs.c [hunk 5]
```diff
@@ -255,6 +256,7 @@ int iotjs_entry(int argc, char** argv) {
   }
 
   // Release JerryScript engine.
+  jerryx_close_handle_scope(jerryx_handle_scope_get_root());
```
Comment (yorkie) [inline src/iotjs.c]: The same to this line :)

File: src/iotjs.c (hunk 4 above)
Comment (legendecas) [reply src/iotjs.c]: Once NAPI module are introduced to ShadowNode, handle-scope are necessary rather than configurable.

**Pull request merged**